### DISCUSSION TITLE
how to communicate between main process with vite.config.js of renderer folder?

### DISCUSSION THREAD

**Discussion by FoxDaxian** [cawa-93/vite-electron-builder#561] [Q&A]
I wanna send info to main process from `vite.config.js` of renderer folder use `ipcRenderer.send()`, but it is undefined~
I have no idea to process this situation. so if anybody dealt with this situation. give me some response please~

**Comment by FoxDaxian** [comment]
@cawa-93 sorry to disturb you~

**Comment by cawa-93** [comment]
There is no way. `vite.config.js` is a configuration that is read and processed exclusively in *compile time* and does not exist in *run time*. At the same time, `ipcRenderer` is a module that provides communication between the main and the renderer in the *run time*.

I see two ways depending on your task:
- Make what you want to share in a separate file and just import it into Main and renderer.
- Transfer what you want to share to the renderer src and send from there using IPC

**Reply by FoxDaxian** [reply]
in my case. i wanna fetch cached data from some `local-db` from main process when the rollup load hooks is invoked. and then. I can change `Vue sfc` content `dynamic`.
as you mention above. Is any other way to do this beside `IPC` communication?

**Reply by FoxDaxian** [reply]
I even thought about using http request to get info...
but it is too slow..

**Reply by cawa-93** [reply]
If you want use "rollup load hooks" then you should create plugin for vite. It fetch your data and modifi vue sfc, or pass it to [`define`](https://vitejs.dev/config/#define)

**Reply by FoxDaxian** [reply]
sorry. not this mean~
but like this:
![image](https://user-images.githubusercontent.com/19708789/141778945-fdc0f824-b93d-42b7-aa4e-f5e8654d19a9.png)

**Reply by cawa-93** [reply]
Try to create function-wrapper in preload [like this](https://github.com/cawa-93/anime-library/blob/main/packages/preload/src/WindowControls/index.ts). And use it like `window.wrapperFunc(...)`

**Comment by FoxDaxian** [comment]
😢
